angularjs-portal-home/src/main/webapp/my-app/layout/widget/controllers.js

```diff
@@ -2,163 +2,194 @@
-                            $scope.isEmpty = true;
-                        }
-                    } else {
-                        console.warn("Got nothing back from widget fetch from " + $scope.portlet.widgetURL);
-                        $scope.isEmpty = true;
-                    }
-                });
-            }
-        };
-      
-        $scope.filteredArray = function (array, objectVar, strings) {
-          return array.filter(function (letter) {
-            for(var i = 0; i < strings.length ; i++) {
-              if(letter[objectVar].indexOf(strings[i]) != -1) {
-                return true;
-              }
-            }
-          });
-        };
+	app.controller('WeatherController', ['$scope', 'layoutService', function($scope, layoutService){
+		$scope.weatherData = [];
+		$scope.loading = false;
+		$scope.showMetric = false;
+		$scope.tempInC1 = 0;
+		$scope.tempInC2 = 0;
+		$scope.celsiusForecasts = [];
+		$scope.dailyHigh = 0;
+		$scope.dailyLow = 0;
+		$scope.currentlyImperial = true;
+		var populateWidgetContent = function() {
+			if($scope.portlet.widgetURL && $scope.portlet.widgetType) {
+				$scope.loading = true;
+				//fetch portlet widget json
+				$scope.portlet.widgetData = [];
+				layoutService.getWidgetJson($scope.portlet).then(function(data) {
+					$scope.loading = false;
+					if(data) {
+						console.log(data);
+						$scope.portlet.widgetData = data.weathers;
+						$scope.weatherData = $scope.portlet.widgetData;
+					} else {
+						$scope.error = true;
+						console.warn("Got nothing back from widget fetch");
+					}
+				}, function(){
+					$scope.loading = false;
+					$scope.error = true;
+				});
+			}
+		};
+		$scope.changeToC = function(){
+			if ($scope.currentlyImperial){
+				for (var i = 0; i < $scope.weatherData.length; i++){
+					$scope.weatherData[i].currentWeather.temperature = ($scope.weatherData[i].currentWeather.temperature-32)*(5/9);
+					
+					for (var j=0; j <$scope.weatherData[i].forecast.length; j++){
+						$scope.weatherData[i].forecast[j].highTemperature = ($scope.weatherData[i].forecast[j].highTemperature-32)*(5/9)
+						$scope.weatherData[i].forecast[j].lowTemperature = ($scope.weatherData[i].forecast[j].lowTemperature-32)*(5/9)
+					}
+				}
+				$scope.currentlyImperial = false;
+			}
+		};
+		$scope.changeToF = function(){
+			if (!$scope.currentlyImperial){
+				for (var i = 0; i < $scope.weatherData.length; i++){
+					$scope.weatherData[i].currentWeather.temperature = ($scope.weatherData[i].currentWeather.temperature)*(9/5) +32;
+					
+					for (var j=0; j <$scope.weatherData[i].forecast.length; j++){
+						$scope.weatherData[i].forecast[j].highTemperature = ($scope.weatherData[i].forecast[j].highTemperature)*(9/5) + 32;
+						$scope.weatherData[i].forecast[j].lowTemperature = ($scope.weatherData[i].forecast[j].lowTemperature)*(9/5) + 32;
+					}
+				}
+				$scope.currentlyImperial = true;
+			}
+		}
```

Comment:
Curious if there is a way we just have a "toggle" function instead of having 2 functions that do pretty much the same thing.